src/lib/transaction_snark/test/account_timing/account_timing.ml

```diff
@@ -837,4 +813,601 @@ let%test_module "account timing check" =
+                Timed
+                  { initial_minimum_balance =
+                      Currency.Balance.of_int 100_000_000_000_000
+                  ; cliff_time = Mina_numbers.Global_slot.of_int 10000
+                  ; cliff_amount = Currency.Amount.of_int 100_000_000_000_000
+                  ; vesting_period = Mina_numbers.Global_slot.of_int 1
+                  ; vesting_increment = Currency.Amount.of_int 10
+                  }
+              in
+              let balance_as_amount = Currency.Balance.to_amount balance in
+              (keypair, balance_as_amount, nonce, timing))
+          |> Array.of_list
+        in
+        (* min balance = balance, spending anything before cliff should trigger min balance violation *)
+        let parties_command =
+          let open Mina_base in
+          let fee = Currency.Fee.of_int 1_000_000 in
+          let amount = Currency.Amount.of_int 10_000_000_000_000 in
+          let nonce = Account.Nonce.zero in
+          let memo =
+            Signed_command_memo.create_from_string_exn
+              "Snapp transfer, timed account"
+          in
+          let sender_keypair = List.hd_exn keypairs in
+          let zkapp_keypair = List.nth_exn keypairs 1 in
+          let receiver_key =
+            zkapp_keypair.public_key |> Signature_lib.Public_key.compress
+          in
+          let (parties_spec : Transaction_snark.For_tests.Spec.t) =
+            { sender = (sender_keypair, nonce)
+            ; fee
+            ; receivers = [ (receiver_key, amount) ]
+            ; amount
+            ; snapp_account_keypairs = []
+            ; memo
+            ; new_snapp_account = false
+            ; snapp_update = Party.Update.dummy
+            ; current_auth = Permissions.Auth_required.Signature
+            ; call_data = Snark_params.Tick.Field.zero
+            ; events = []
+            ; sequence_events = []
+            }
+          in
+          Transaction_snark.For_tests.multiple_transfers parties_spec
+        in
+        return (ledger_init_state, parties_command)
+      in
+      Quickcheck.test ~seed:(`Deterministic "zkapp command, just before cliff")
+        ~sexp_of:[%sexp_of: Mina_ledger.Ledger.init_state * Parties.t] ~trials:1
+        gen ~f:(fun (ledger_init_state, parties) ->
+          Mina_ledger.Ledger.with_ephemeral_ledger
+            ~depth:constraint_constants.ledger_depth ~f:(fun ledger ->
+              Mina_ledger.Ledger.apply_initial_ledger_state ledger
+                ledger_init_state ;
+              let _state_body, state_view =
+                state_body_and_view_at_slot
+                  Mina_numbers.Global_slot.(of_int 9999)
+              in
+              match
+                Mina_ledger.Ledger.apply_parties_unchecked ~constraint_constants
+                  ~state_view ledger parties
+              with
+              | Ok _txn_applied ->
+                  failwith "Should have failed with min balance violation"
```

Comment:
Same as above

Reply:
added a failure case "while vesting", factored out common code for checking failure status.